cli/crates/backend/src/project.rs

```diff
@@ -181,12 +167,38 @@ pub async fn init(name: Option<&str>, template: Template<'_>) -> Result<(), Back
 
             schema_write_result?;
             dot_env_write_result?;
+
+            // This has to be the last thing we do.
+            if let ConfigType::TypeScript = config_type {
+                add_dependencies(&project_path)?;
+            };
         }
     };
 
     Ok(())
 }
 
+fn add_dependencies(project_path: &Path) -> Result<(), BackendError> {
+    environment::add_dev_dependency_to_package_json(
+        project_path,
+        GRAFBASE_SDK_PACKAGE_NAME,
+        GRAFBASE_SDK_PACKAGE_VERSION,
+    )
+    .map_err(BackendError::CommonError)?;
+
+    let package_manager = environment::guess_project_package_manager(project_path, &project_path.join("package.json"))
+        .unwrap_or_default();
+
+    if let JavaScriptPackageManager::Npm = package_manager {
+        environment::install_dependencies(project_path).map_err(BackendError::CommonError)?;
+    } else {
+        let error = io::Error::new(io::ErrorKind::InvalidInput, "Project is not using npm.");
+        return Err(BackendError::CommonError(CommonError::NpmNotFound(Box::new(error))));
```

Comment:
I'd duplicate this error rather than emitting a common one that isn't coming from common